src/address.ts

```diff
@@ -140,7 +165,7 @@ export class Address {
         const prefix = decoded?.prefix;
         const pubkey = decoded ? Buffer.from(bech32.fromWords(decoded.words)) : undefined;
 
-        if (prefix !== HRP || pubkey?.length !== PUBKEY_LENGTH) {
+        if (prefix !== LibraryConfig.DefaultAddressHrp || pubkey?.length !== PUBKEY_LENGTH) {
```

Comment:
why is the address considered not valid if the hrp is different than the default?

Reply:
Good question!

In order to avoid a breaking change with respect to the behavior of `isValid` (public method). We will change this in the future (a major version).